src/ServerHandler.cpp

```diff
@@ -129,25 +139,53 @@ void ServerHandler::parseMessageFrame(Frame messageFrame) {
 
 void ServerHandler::parseServerFrame() {
     std::string frameString;
-    _connectionHandler->getFrameAscii(frameString, STOMP_DELIMITER.c_str()[0]);
+    while(_connectionHandler == nullptr || !_connectionHandler->getFrameAscii(frameString, STOMP_DELIMITER.c_str()[0])) {
+        if(_shouldTerminate) return;
+    }
     Frame frame = Frame(frameString);
     FrameCommand cmd = frame.getCommand();
+    if(!_loggedIn && cmd == RECEIPT) {
+        delete _connectionHandler;
+        _connectionHandler = nullptr;
+        sendFrameToClient(frame);
+    }
```

Comment:
why is that?

Reply:
We want to get the receipt for the disconnect before actually disconnecting

Reply:
ok, now i saw where it is. good job